src/ui/ui-storyEditor.js

```diff
@@ -252,4 +293,107 @@ https://raw.githubusercontent.com/fluid-project/sjrk-story-telling/master/LICENS
         that.events.onRemoveBlocksCompleted.fire(removedBlockKeys);
     };
 
+    /**
+     * Reorders the specified block UI component in the specified direction
+     *
+     * @param {Component} reorderer - an instance of fluid.reorderList
+     * @param {jQuery} blockUi - a jQueryable of an sjrk.storyTelling.blockUi.editor's container
+     * @param {Number} direction - a member of fluid.direction
+     */
+    sjrk.storyTelling.ui.storyEditor.reorderBlock = function (reorderer, blockUi, direction) {
+        var relativePosition = reorderer.layoutHandler.getRelativePosition(blockUi, direction);
+
+        reorderer.requestMovement(relativePosition, blockUi);
+    };
+
+    /**
+     * Updates the order of each block model according to the order of the blockUi
+     * elements in the story content editing area
+     *
+     * @param {Component} that - an instance of sjrk.storyTelling.ui.storyUi
+     */
+    sjrk.storyTelling.ui.storyEditor.updateBlockOrder = function (that) {
+        var blockUis = that.reorderer.container.children();
+
+        // used to retrieve the class name for each block
+        // within the blockManager's managedViewComponentRegistry
+        var managedClassNamePattern = sjrk.storyTelling.ui.storyEditor.getManagedClassNamePattern(that.blockManager.options.selectors.managedViewComponents);
+
+        for (var i = 0; i < blockUis.length; i++) {
+            var managedClassName = fluid.find(blockUis[i].classList, function (value) {
+                if (typeof value === "string") {
+                    var patternMatches = value.match(managedClassNamePattern);
+                    return patternMatches === null ? undefined : patternMatches.input;
+                }
+            });
+
+            var block = that.blockManager.managedViewComponentRegistry[managedClassName].block;
+
+            var firstInOrder = i === 0 ? true : false;
```

Comment:
Line 332-341 can be simplified to:

```
block.applier.change({
    path: "",
    value: {
        order: i,
        firstInOrder: i === 0 ? true : false;
        lastInOrder: i === blockUis.length - 1 ? true : false;
    }
});
```

Reply:
This doesn't seem to work: Editor tests are failing and it breaks the reorderer buttons. I'll continue to investigate.

Reply:
This form works: 

```
block.applier.change("", {
  order: i,
  firstInOrder: i === 0 ? true : false,
  lastInOrder: i === blockUis.length - 1 ? true : false
});
```

Reply:
@cindyli 's code should be identical in all observable effects - I'd be interested in reason for the discrepancy once you find it.
Note that the expression can be simplified yet further:
````
block.applier.change({
    path: "",
    value: {
        order: i,
        firstInOrder: i === 0,
        lastInOrder: i === blockUis.length - 1
    }
});
````
(@cindyli 's version had a few stray semicolons but I imagine you removed them)

Reply:
Removing the `path` and `value` references did the trick. Thanks @cindyli and @amb26 :)

Reply:
> > > When dragging the mouse to select some buttons, the "play" button from the text-to-speech shows up. Clicking on it doesn't read anything. If this is a problem, it probably should be addressed in a separate JIRA.
> > > <img alt="Screen Shot 2020-06-23 at 9 20 53 PM" width="340" src="https://user-images.githubusercontent.com/572877/85486044-e121cf80-b597-11ea-9432-7d1e91da22a3.png">
> > 
> > 
> > We still haven't patched in the fix for FLUID-6512, which would eliminate the page reload when `Play` is clicked, but I'm not sure it's a problem that an empty selection isn't readying anything. The Orator is not designed to read anything other than what is displayed on the screen, if I recally correctly, so no alt-text or aria things. @jobara may be able to confirm this behaviour.
> 
> The Orator only reads the visible text. I suppose you should be able to configure that behaviour although it would likely require some new functions be implemented to know how to parse the text.
> 
> The play button shows up with a selection, if you are finding that it shows up even when no text is selected, perhaps a JIRA against the Orator is warranted.

This has been captured in [FLUID-6523](https://issues.fluidproject.org/browse/FLUID-6523)